core/src/main/java/net/opentsdb/meta/DefaultBatchMetaQuery.java

```diff
@@ -91,6 +91,12 @@ protected DefaultBatchMetaQuery(final Builder builder) {
       end = new MillisecondTimeStamp(
               DateTime.parseDateTimeString(builder.end, builder.time_zone));
     }
+
+    if (type == QueryType.NAMESPACES) {
+      if (meta_query.size() > 1) {
+        throw new IllegalArgumentException("Namespaces query cannot have more than one query");
```

Comment:
Keep an eye on this. We may have use cases where someone has multiple filters for namespace.